src/adapters/sharethrough.js

```diff
@@ -0,0 +1,149 @@
+var utils = require('../utils.js');
+var bidmanager = require('../bidmanager.js');
+var bidfactory = require('../bidfactory.js');
+
+const STR_BIDDER_CODE = "sharethrough";
+const STR_VERSION = "0.1.0";
+
+var SharethroughAdapter = function SharethroughAdapter() {
+
+  const str = {};
+  str.STR_BTLR_HOST = document.location.protocol + "//btlr.sharethrough.com";
+  str.STR_TEST_HOST =  document.location.protocol + "//btlr-prebid-test.sharethrough.com";
+  str.STR_BEACON_HOST = document.location.protocol + "//b.sharethrough.com/butler?";
+  str.placementCodeSet = {};
+
+  function _callBids(params) {
+    const bids = params.bids;
+
+    $$PREBID_GLOBAL$$.onEvent('bidWon', str.bidWon);
+
+    addEventListener("message", _receiveMessage, false);
+    
+    // cycle through bids
+    for (let i = 0; i < bids.length; i += 1) {
+      const bidRequest = bids[i];
+      str.placementCodeSet[bidRequest.placementCode] = bidRequest;
+      const scriptUrl = _buildSharethroughCall(bidRequest);
+      str.loadIFrame(scriptUrl);
```

Comment:
We need to put the url inside an iframe so we can drop tracking cookies on publisher's page. This method work for iOS Safari since they block 3rd party cookies.